### ISSUE TITLE
Add test data and checks for read10xResults()

### ISSUE THREAD

**Issue by davismcc** [alanocallaghan/scater#5]

**Comment by LuyiTian** [comment]
I found the sce object generated by read10xResults is not compatible with many scater and scran methods. because the counting matrix will be dgTMatrix:

```
class(counts(sce))
[1] "dgTMatrix"
attr(,"package")
[1] "Matrix"
```

but normally we have matrix:

```
class(counts(sce))
[1] "matrix"
```

Then some functions will crash because of this:

```
sce_qc <- calculateQCMetrics(sce)
Error in .get_qc_metrics_per_cell(exprs_mat, exprs_type = exprs_values,  : 
  dgTMatrix not supported, convert to dgCMatrix
```

**Comment by LTLA** [comment]
This shouldn't be the case, see line 77 of `10ximport-wrapper.R` for an explicit conversion to a `dgCMatrix`. I would have thought that this would have made it to the release version.

**Comment by babiddy** [comment]
I have had problems with this function as well. Specifically, when trying to load multiple 10x datasets I recieve the following error:

`Error in rownames<-(*tmp*, value = .get_colnames_from_assays(assays)) : duplicate rownames not allowed`

This does not occur if I load each dataset separately. 

Also when trying to use the `min_total_cell_counts` or `min_mean_gene_counts` arguments I receive this error:

`Error in .general_colSums(data_mat) : dgTMatrix not supported, convert to dgCMatrix`

Thanks, for the help. Love scater.

Brent

**Comment by LTLA** [comment]
These bugs should be fixed with #34.

**Comment by MrChenFeng** [comment]
still wrong

**Comment by LTLA** [comment]
Perhaps you could be more informative. Like, the version of _scater_ being used? The error message? The code you're using to generate the error message? Keep in mind #34 hasn't made it to release.

**Comment by MrChenFeng** [comment]
uh, maybe when ur patch uploaded it will be ok! 
i write a simple fuction to load data and its ok 4 me
thank u!

**Comment by LTLA** [comment]
`read10xResults` will be replaced by `DropletUtils::read10xCounts` in the next release, which is probably a more suitable home for a technology-specific data import function.

**Comment by koelling** [comment]
I just ran into the same problem using scater 1.6.1. Given that the latest version of scater has this bug and that DropletUtils doesn't seem to be released yet, what would be the best way of importing 10X data into scater right now?

Use Seurat::Read10X and then manually set up a SingleCellExperiment class?

**Comment by LTLA** [comment]
Get @davismcc to update the release branch on BioC, as I don't have write access to that.

Davis, this will entail manually copying the version of `10x-import-wrapper.R` to `RELEASE_3_6`, prior to my latest changes (i.e., just before I replaced the function with the `DropletUtils::` version).

**Comment by koelling** [comment]
That would be great, thank you! If that is too complicated maybe it would also be an option to install a patched version of scater through devtools somehow? Not sure which commit/branch I would need to use for that though...

**Comment by LTLA** [comment]
Propagated stop-gap solution in 57c6689c34b323c198189130dc62d37e62d643a0. Needs push to Bioc from @davismcc (also push `master` to BioC-devel while you're at it). Will close this issue as soon as that happens.

**Comment by LTLA** [comment]
Pushed to BioC.

**Comment by meichendong** [comment]
Hi, I'm trying to use R 3.4.3 to do analysis and my scater package is of the version ` scater_1.6.3`, I got error when I run `calculateQCMetrics` function.
My code was like:
`sceset <- SingleCellExperiment(assays = list(counts = sc_counts), 
                    colData = sc_pdata, rowData = sc_fdata)
rowData(sceset)$Mito <- grepl("MT-", rowData(sceset)$symbol)
sceset <- calculateQCMetrics(sceset, feature_controls = list(mt = rowData(sceset)$Mito))`

The error was like LuyiTian mentioned above, I got:
`Error in .get_qc_metrics_per_cell(exprs_mat, exprs_type = exprs_values,  : 
  dgTMatrix not supported, convert to dgCMatrix`

I don't quite understand how the `10ximport-wrapper.R` code would solve this problem. 
I'm using the public data from the 10xgenomics websites: https://support.10xgenomics.com/single-cell-gene-expression/datasets/2.0.1/pbmc4k
Since I already have the SingleCellExperiment object, do I have to go back and use read10xResults()?

Thanks!
